packages/react-aria-components/src/Collection.tsx

```diff
@@ -916,72 +103,18 @@ export const Section = /*#__PURE__*/ createBranchComponent('section', <T extends
   return render(props, ref, section);
 });
 
-export const CollectionContext = createContext<CachedChildrenOptions<unknown> | null>(null);
-
-/** A Collection renders a list of items, automatically managing caching and keys. */
-export function Collection<T extends object>(props: CollectionProps<T>): JSX.Element {
-  let ctx = useContext(CollectionContext)!;
-  props = mergeProps(ctx, props);
-  props.dependencies = (ctx?.dependencies || []).concat(props.dependencies);
-  let children = useCollectionChildren(props);
-
-  let doc = useContext(CollectionDocumentContext);
-  if (doc) {
-    return <CollectionRoot>{children}</CollectionRoot>;
-  }
-
-  return <>{children}</>;
-}
-
-function CollectionRoot({children}) {
-  let doc = useContext(CollectionDocumentContext);
-  let wrappedChildren = useMemo(() => (
-    <CollectionDocumentContext.Provider value={null}>
-      <ShallowRenderContext.Provider value>
+export function useClearCollectionRenderer(children: ReactNode) {
```

Comment:
this feels weird that a hook is somehow clearing a context which requires rendering jsx
it's private, so I'm not super fussed about it, but I don't see anyplace that we use it where it couldn't be a component instead

Reply:
Yeah it could be. I guess at that point we might as well just always set the context even if it is already the default. Would result in the same tree depth. I was just trying to avoid an extra element when it wasn't needed